### PR TITLE
Dmp 760 delete requested and processed audio

### PR DESCRIPTION
**Pull request by garethmorriscgi**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DMP-760

### Change description ###
Services endpoint that handles the deletion of audio request details and if present any audio blob data from outbound storage container and the linking transient object

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/resources/openapi/audio.yaml [hunk 1]
```diff
@@ -29,6 +29,31 @@ paths:
...
         description: Audio Request item to add
+    delete:
+      tags:
+        - Audio
+      summary: Delete Audio Request and Processed Audio
+      description: Delete Audio Request and Processed Audio
+      operationId: deleteAudioRequest
+      parameters:
+        - in: query
+          name: audioRequestId
```
Comment (richardbull) [inline src/main/resources/openapi/audio.yaml:40]: Query params should be lower snake case
but I think this should really be a path param to be similar to `DELETE
[/courthouses/{courthouse_id}](http://localhost:4550/swagger-ui/index.html#/Courthouses/courthousesCourthouseIdDelete)`

File: src/main/resources/openapi/audio.yaml [hunk 2]
```diff
@@ -29,6 +29,31 @@ paths:
             schema:
               $ref: '#/components/schemas/AudioRequestDetails'
         description: Audio Request item to add
+    delete:
```
Comment (richardbull) [inline src/main/resources/openapi/audio.yaml:32]: I think this endpoint should go along the new one I added this sprint
[audio-requests.yaml](https://github.com/hmcts/darts-api/blob/master/src/main/resources/openapi/audio-requests.yaml)

DELETE `/audio-requests/{id}` ?

FYI @hemantasharma1129

File: src/main/resources/openapi/audio.yaml [hunk 3]
```diff
@@ -29,6 +29,31 @@ paths:
...
+      description: Delete Audio Request and Processed Audio
+      operationId: deleteAudioRequest
+      parameters:
+        - in: query
+          name: audioRequestId
+          required: true
+          schema:
+            $ref: '#/components/schemas/AudioRequestId'
+      responses:
+        '200':
```
Comment (richardbull) [inline src/main/resources/openapi/audio.yaml:45]: I was expecting a `204` response code - but the ticket does have `200`. Maybe worth checking with @hemantasharma1129

File: src/main/resources/openapi/audio.yaml [hunk 4]
```diff
@@ -29,6 +29,31 @@ paths:
...
+          description: audio request has been deleted
+        '500':
```
Comment (richardbull) [inline src/main/resources/openapi/audio.yaml:47]: Do we need to specify `500` response - what errors do you have coming back, is `400` Bad Request missing?

File: src/main/java/uk/gov/hmcts/darts/audio/controller/AudioController.java [hunk 5]
```diff
@@ -38,6 +38,12 @@ public ResponseEntity<Void> addAudioRequest(AudioRequestDetails audioRequestDeta
         return new ResponseEntity<>(HttpStatus.CREATED);
     }
 
+    public ResponseEntity<Void> deleteAudioRequest(Integer audioRequestId) {
```
Comment (richardbull) [inline src/main/java/uk/gov/hmcts/darts/audio/controller/AudioController.java:41]: If Hemanta agrees with the base path being `audio-requests` pluralised - then I think the new services we do this sprint should go under `AudioRequestsController`

File: src/test/java/uk/gov/hmcts/darts/audio/service/impl/MediaRequestServiceImplTest.java [hunk 6]
```diff
@@ -5,18 +5,25 @@
...
+import uk.gov.hmcts.darts.common.repository.TransientObjectDirectoryRepository;
 import uk.gov.hmcts.darts.common.repository.UserAccountRepository;
+import uk.gov.hmcts.darts.datamanagement.api.impl.DataManagementApiImpl;
 
 import java.time.OffsetDateTime;
+import java.util.Optional;
+import java.util.UUID;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.eq;
```
Comment (richardbull) [inline src/test/java/uk/gov/hmcts/darts/audio/service/impl/MediaRequestServiceImplTest.java:26]: Can you check SonarQube - I have a feeling I did this style before and spotted it was a code smell
https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&sinceLeakPeriod=true&pullRequest=379&id=uk.gov.hmcts.reform%3Adarts-api

File: src/test/java/uk/gov/hmcts/darts/audio/service/impl/MediaRequestServiceImplTest.java [hunk 7]
```diff
@@ -90,4 +100,51 @@ void whenSavingAudioRequestIsSuccessful() {
...
+
+    @Test
+    void whenNoAudioIsPresentOnlyDeleteAudioRequest() {
+        var mediaRequestId = 1;
+        when(transientObjectDirectoryRepository.getTransientObjectDirectoryEntityByMediaRequest_Id(mediaRequestId))
+            .thenReturn(Optional.empty());
+
+        mediaRequestService.deleteAudioRequest(mediaRequestId);
+
+        verify(mockMediaRequestRepository, Mockito.times(1)).deleteById(eq(mediaRequestId));
```
Comment (richardbull) [inline src/test/java/uk/gov/hmcts/darts/audio/service/impl/MediaRequestServiceImplTest.java:145]: I think 1 times is the default, so can be simplified FYI

File: src/main/resources/openapi/audio.yaml (hunk 2 above)
Comment (hemantasharma1129) [reply src/main/resources/openapi/audio.yaml:32]: It should be part of AudioController which has endpoints for requesting an Audio, downloading an Requested Audio etc. AudioRequestController may no longer be required? We can discuss about it.

File: src/main/java/uk/gov/hmcts/darts/audio/service/impl/MediaRequestServiceImpl.java [hunk 8]
```diff
@@ -74,6 +80,26 @@ public Integer saveAudioRequest(AudioRequestDetails request) {
...
+            UUID blobId = mediaTransientObject.getExternalLocation();
+
+            if (blobId != null) {
+                dataManagementApi.deleteBlobDataFromOutboundContainer(blobId);
+            }
+
+            transientObjectDirectoryRepository.deleteById(mediaTransientObject.getId());
+        }
+
+        mediaRequestRepository.deleteById(mediaRequestId);
```
Comment (hemantasharma1129) [inline src/main/java/uk/gov/hmcts/darts/audio/service/impl/MediaRequestServiceImpl.java:100]: what happens if the requestId which we are trying to delete doesnt exist in Media Request table? Which error the JPA repository thows?

**Review by hemantasharma1129 — APPROVED** [review]

**Pull request merged**